File: submissions/MohanKumarS/phase2/day35_stack.c
#include <stdio.h>
#define SIZE 20

int stack[SIZE];
int top = -1;

void push(int(value))
{
    if(top == SIZE - 1)
    {
        printf("Stack Overflow");
        return;
    }   
    else
    {
        top = top + 1;
        stack[top] = value;
    }
}

void pop()
{
    if(top == -1)
    {
        printf("Stack Underflow");
        return;
    }
    else
        top = top - 1;
}

void display()
{
    if (top == -1) 
    {
        printf("Stack is empty.\n");
        return;
    }
    for (int i = 0; i <= top; i++) 
    {
        printf("%d ", stack[i]);
    }
    printf("\n");
}

int main()
{
    push(100);
    push(200);
    push(300);
    printf("Before pop: ");
    display();
    pop();
    printf("After pop: ");
    display();

    return 0;
}

File: submissions/Harini S/DAY 35/day35_stack.py
class Stack:
    def __init__(self,size):
        self.size=size
        self.stack=[None]*size
        self.tos=-1
    def push(self,value):
        if self.tos>=self.size-1: print("Stack overflow")
        else:
            self.tos+=1
            self.stack[self.tos]=value
    def pop(self):
        if self.tos==-1: print("Stack underflow")
        else:
            self.stack[self.tos]=None
            self.tos-=1
    def display(self):
        if self.tos==-1: print("stack underflow")
        else:
            print("Current Stack:",end=" ")
            for i in self.stack[:self.tos+1]: print(i,end=" ")
lis=list(map(int,input("Input: ").split()))
pop_value=int(input("Enter pop value:"))
s=Stack(len(lis))
for i in lis: s.push(i)
for i in range(pop_value): s.pop()
s.display()